defold-spine/src/comp_spine_model.cpp

```diff
@@ -858,32 +934,47 @@ namespace dmSpine
         }
         else if (params.m_PropertyId == PROP_ANIMATION)
         {
-            out_value.m_Variant = dmGameObject::PropertyVar(component->m_AnimationId);
+            dmhash_t value = 0;
+            IF_CAN_GET_TRACK()
```

Comment:
I'm sorry, this is not very readable to me (it's _very_ hard to see what values are used and to inspect the code).
It will also be fairly undebuggable.
If you wish to simplify the code, please use functions (or simple expressions):

> if (!IsTrackIndexValid(component, trackIndex) 
> return dmGameObject::PROPERTY_RESULT_INVALID_INDEX

> static Track* GetTrackFromIndex(component, trackIndex);
> Track* track = GetTrackFromIndex(component, trackIndex);
> if (track) ...

etc